Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1

```diff
@@ -1625,6 +1625,305 @@ function Test-DiscoveryService{
+      }
+     }
+    }
+    $azureInstallationConfig.Save("$azureConfigPath") | Out-Null
+}
+
+function Get-XMLChildNode {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string] $installConfigPath,
+        [Parameter(Mandatory=$true)]
+        [string] $configSection,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeGetAttribute,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeAttributeSetting
+    )
+
+    $installationConfig = [xml](Get-Content $installConfigPath)
+    $identityScope = $installationConfig.installation.settings.scope | Where-Object {$_.name -eq $configSection}
+    $existingChildNode = @{}
+
+    $existingChildNode = $identityScope.ChildNodes | Where-Object {$_.$childNodeGetAttribute -eq $childNodeAttributeSetting}
+    if ($null -eq $existingChildNode)
+    {
+      Write-DosMessage -Level "Information" -Message "$childNodeAttributeSetting not found in install.config"
+    }
+    return $existingChildNode
+}
+
+function Set-XMLChildNode {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string] $azureConfigPath,
+        [Parameter(Mandatory=$true)]
+        [string] $configSection,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeGetAttribute,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeSetAttribute,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeAttributeSetting,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeAttributeValue
+    )
+
+    $azureInstallationConfig = [xml](Get-Content $azureConfigPath)
+    $azureIdentityScope = $azureInstallationConfig.installation.settings.scope | Where-Object {$_.name -eq $configSection}
+    $existingChildNode = @{}
+
+    $existingChildNode = $azureIdentityScope.ChildNodes | Where-Object {$_.$childNodeGetAttribute -eq $childNodeAttributeSetting}
+    if ($null -ne $existingChildNode)
+    {
+      $existingChildNode.$childNodeSetAttribute = $childNodeAttributeValue
+      $azureInstallationConfig.Save("$azureConfigPath") | Out-Null
+      Write-DosMessage -Level "Information" -Message "$childNodeAttributeValue added to $childNodeAttributeSetting in azureSettings.config"
+    }
+    elseif ($null -eq $existingChildNode)
+    {
+      Write-DosMessage -Level "Information" -Message "$childNodeAttributeSetting not found in azureSettings.config"
+    }
+}
+
+function Get-XMLChildNodes {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string] $installConfigPath,
+        [Parameter(Mandatory=$true)]
+        [string] $configSection,
+        [Parameter(Mandatory=$true)]
+        [string[]] $nodesToSearch,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeGetAttribute
+    )
+
+    $installationConfig = [xml](Get-Content $installConfigPath)
+    $identityScope = $installationConfig.installation.settings.scope | Where-Object {$_.name -eq $configSection}
+    $allExistingChildNodes = @()
+    $count = 1
```

Comment:
I have to add a place holder for each node, so I know when one ends and another one starts. I might be able to leave off the count, since I'm not using that in the check.